de-lib/src/main/java/org/iplantc/de/diskResource/client/views/toolbar/DiskResourceViewToolbarImpl.java

```diff
@@ -495,7 +498,13 @@ void onMoveToTrashClicked(SelectionEvent<Item> event) {
 
     @UiHandler("newFolderMi")
     void onNewFolderClicked(SelectionEvent<Item> event) {
-        presenter.onCreateNewFolderSelected(selectedFolder);
+        if(selectedFolder == null) {
+            Folder parent = drAbFactory.folder().as();
```

Comment:
I think we're trying to keep AutoBean stuff out of the views.  Could we move this new logic to the presenter (and update/add tests)?